GalilSup/src/GalilController.cpp

```diff
@@ -3164,9 +3172,20 @@ void GalilController::GalilStartController(char *code_file, int burn_program, in
 
 		//Remove the \r characters
 		dc.erase (std::remove(dc.begin(), dc.end(), '\r'), dc.end());
+		
+		int code_changed = compareCode(dc, uc) && dc.compare("") != 0;
+		
+		if (code_changed || !quiet_start) {
+			stopThreads();
+			stopAxes();
+		}
+		else
+		{
+			printf("\nControl code and quiet start requested. Threads will not be restarted\n");
+		}
 
 		/*If code we wish to download differs from controller current code then download the new code*/
-		if (compareCode(dc, uc) && dc.compare("") != 0)
+		if (code_changed)
```

Comment:
The previous logic stopped all threads and then started thread 0, which was assumed to start all other threads. The quiet_start will attempt to start thread 0, which is likely to be running. What happens if thread 0 is still running but the other threads have died, will they get started again by this call? If not, then a call to checkGalilThreads() should be added and if that fails then thread 0 should be restarted. It would need to be confirmed that this works ok if a subset of expected threads has stopped rather than all of them.